main.tf

```diff
@@ -0,0 +1,146 @@
+data "aws_caller_identity" "current" {}
+
+data "aws_region" "current" {}
+
+locals {
+  account_role_prefix  = coalesce(var.account_role_prefix, "${var.cluster_name}-account")
+  operator_role_prefix = coalesce(var.operator_role_prefix, "${var.cluster_name}-operator")
+  sts_roles = {
+    installer_role_arn    = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${local.account_role_prefix}-Installer-Role",
+    support_role_arn      = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${local.account_role_prefix}-Support-Role",
+    controlplane_role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${local.account_role_prefix}-ControlPlane-Role",
+    worker_role_arn       = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${local.account_role_prefix}-Worker-Role"
```

Comment:
If `var.create_account_roles` true, take those values from the module output, otherwise build it manually
This way you will also get the dependency

Reply:
What about this comment? lets have this with the right dependency